catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/LoopBrickTest.java

```diff
@@ -0,0 +1,156 @@
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.BrickTestUtils.checkIfBrickAtPositionShowsString;
+import static org.catrobat.catroid.uiespresso.content.brick.BrickTestUtils.onScriptList;
+
+@RunWith(AndroidJUnit4.class)
+public class LoopBrickTest {
+	private int brickPosition;
+	private Script script;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		brickPosition = 1;
+		int timesToRepeat = 3;
+		int changeYMovement = -10;
+
+		LoopEndBrick endBrick = new LoopEndBrick();
+		RepeatBrick repeatBrick = new RepeatBrick(timesToRepeat);
+		repeatBrick.setLoopEndBrick(endBrick);
+
+		script = BrickTestUtils.createProjectAndGetStartScript("LoopBrickTest1");
+		script.addBrick(repeatBrick);
+		script.addBrick(new ChangeYByNBrick(changeYMovement));
+		script.addBrick(endBrick);
+
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void repeatBrickTest() throws InterpretationException {
+
+		checkSetUpBrickArrangement();
+
+		deleteBrickAtPosition(brickPosition);
+
+		checkIfBrickAtPositionShowsString(1, R.string.brick_change_y_by);
+		onView(withText(R.string.brick_loop_end))
+				.check(doesNotExist());
+
+		addLoopAtPosition(R.string.brick_repeat, brickPosition);
+
+		// selects block and drags it from Top_Center (of the block) to the bottom of the device screen
+		onScriptList().atPosition(2)
+				.perform(longClick()).perform(new GeneralSwipeAction(Swipe.FAST,
+				GeneralLocation.TOP_CENTER, new CoordinatesProvider() {
+					@Override
+					public float[] calculateCoordinates(View view) {
+						float[] coordinates = GeneralLocation.CENTER.calculateCoordinates(view);
+						coordinates[1] = view.getContext().getResources().getDisplayMetrics().heightPixels;
+						return coordinates;
+					}
+				},
+				Press.FINGER));
+
+		checkSetUpBrickArrangement();
+	}
+
+	@Test
+	public void foreverBrickTest() throws InterpretationException {
+
+		addLoopAtPosition(R.string.brick_forever, 2);
+
+		checkIfBrickAtPositionShowsString(1, R.string.brick_repeat);
+		checkIfBrickAtPositionShowsString(2, R.string.brick_forever);
+		checkIfBrickAtPositionShowsString(3, R.string.brick_change_y_by);
+		checkIfBrickAtPositionShowsString(4, R.string.brick_loop_end);
+		checkIfBrickAtPositionShowsString(5, R.string.brick_loop_end);
+
+		deleteBrickAtPosition(2);
+
+		checkSetUpBrickArrangement();
+	}
+
+	public void deleteBrickAtPosition(int position) {
+		onScriptList().atPosition(position)
+				.perform(click());
+		onView(withText(R.string.brick_context_dialog_delete_brick))
+				.perform(click());
+		onView(withText(R.string.yes))
+				.perform(click());
+	}
+
+	public void addLoopAtPosition(int loopTypIdentifier, int insertPosition) {
```

Comment:
please rename lookTypIdentifier to something like brickHeadterStringId to make more clear what this variable means